File: exs/mundo_1/python/035.py
"""
Desafio 035

Problema: Desenvolva um programa que leia o comprimento de três
          retas e diga ao usuário se elas podem ou não formar um triângulo.

Resolução do problema:
"""
lado_A = float(input('Informe a medida do lado A: '))
lado_B = float(input('Informe a medida do lado B: '))
lado_C = float(input('Informe a medida do lado C: '))
analise = False
#  Somente será validado como Triângulo passando nas três condições seguintes
if abs(lado_B - lado_C) < lado_A < lado_B + lado_C:
    if abs(lado_A - lado_C) < lado_B < lado_A + lado_C:
        if abs(lado_A - lado_B) < lado_C < lado_A + lado_B:
            analise = True
if analise:
    print('As retas PODEM FORMAR um triângulo.')
else:
    print('As retas NÃO PODEM FORMAR um triângulo.')

File: exs/mundo_1/ruby/035.rb
=begin
Desafio 035

Problema: Desenvolva um programa que leia o comprimento de três
          retas e diga ao usuário se elas podem ou não formar um triângulo.

Resolução do problema:
=end

print'Informe a medida do lado A: '
a = gets.chomp.to_f

print'Informe a medida do lado B: '
b = gets.chomp.to_f

print'Informe a medida do lado C: '
c = gets.chomp.to_f

analise = false

if (b-c).abs < 1 and 1 < b + c 
	if (a -c ).abs < b and b < a + c
		if (a - b).abs < c and c < a + b
			analise = true
		end
	end
end
if analise
    puts'As retas PODEM FORMAR um triângulo.'
else
    puts'As retas NÃO PODEM FORMAR um triângulo.'
end